sklearn/utils/tests/test_validation.py

```diff
@@ -1694,6 +1696,37 @@ def test_get_feature_names_pandas():
     assert_array_equal(feature_names, columns)
 
 
+@pytest.mark.parametrize(
+    "dataframe_lib_str, minversion",
+    [("pyarrow", "12.0.0"), ("pandas", "1.5.0"), ("polars", "0.18.2")],
+)
+def test_get_feature_names_dataframe_protocol(dataframe_lib_str, minversion):
+    """Uses the dataframe exchange protocol to get feature names."""
+    dataframe_lib = pytest.importorskip(dataframe_lib_str, minversion=minversion)
+    data = {"col_0": [1, 4], "col_1": [2, 3], "col_2": [3, 6]}
+    df = generate_dataframe_from_lib(dataframe_lib, data)
+    feature_names = _get_feature_names(df)
+
+    assert_array_equal(feature_names, ["col_0", "col_1", "col_2"])
+
+
+def test_is_pandas_df():
+    """Check behavior of is_pandas_df."""
+    pd = pytest.importorskip("pandas")
+    df = pd.DataFrame({"a": [1, 2, 3]})
+    assert _is_pandas_df(df)
+
+    assert not _is_pandas_df(np.asarray([1, 2, 3]))
+    assert not _is_pandas_df(1)
```

Comment:
Do you want to add a test with pyarrow and polars?

Reply:
It seems that this is now tackled as `test_is_pandas_df_other_libraries`.